File: LC/python/Symmetric Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a boolean
    def isSymmetric(self, root):
        if root == None:
            return True
        elif root.left == None and root.right == None:
            return True
        elif root.left != None and root.right!= None:
            return self.symmetric(root.left,root.right)
        else: 
            return False
            
    def symmetric(self,left,right):
        if left == None and right == None:
            return True
        elif left != None and right != None:
            return (left.val == right.val and self.symmetric(left.left, right.right) and self.symmetric(left.right, right.left))
        return False
        

File: LC/java/Symmetric Tree.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root==null) return true;
        
        else if(root.left == null && root.right == null) return true;
        
        else if(root.left != null && root.right != null) return helper(root.left,root.right);
        
        else return false;
    }
    
    private boolean helper(TreeNode x, TreeNode y){
        if(x==null && y== null) return true;
        
        else if (x!=null && y!= null) return x.val == y.val && helper(x.left, y.right) && helper(x.right, y.left);
        
        else return false;
        
        
    }
} 